### PR TITLE
Add BN support to the test infrastructure.

### PR DESCRIPTION
**Pull request by pauli-oracle**
- [x] tests are added or updated

This includes support for:

- comparisions between pairs of BIGNUMs
- comparisions between BIGNUMs and zero
- equality comparison between BIGNUMs and one
- equality comparisons between BIGNUMs and constants
- parity checks for BIGNUMs

### CODE REVIEW

File: test/testutil/tests.c [hunk 1]
```diff
@@ -19,6 +19,9 @@
 #define MEM_BUFFER_SIZE     (33)
 #define MAX_STRING_WIDTH    (80)
 
+/* Special representation of -0 */
+static char *const BN_minus_zero = "-0";
```
Comment (levitte) [inline test/testutil/tests.c]: Care to tell me why you placed `const` there?

Comment (pauli-oracle) [inline test/testutil/tests.c]: So the pointer doesn't get changed.

Declaring it as an array is neater.  I'll update.

**Comment by pauli-oracle** [comment]
And updated to an array.

**Comment by levitte** [comment]
I'm like to see a corresponding test suite in `test/test_test.c`

**Comment by pauli-oracle** [comment]
The `test_dec2bn` and `test_hex2bn` do a pretty good job stressing the various calls already, so I didn't include anything in `test_test.c`.  Still, I can add some.

**Comment by pauli-oracle** [comment]
`test_test` tests added.

**Comment by levitte** [comment]
> The `test_dec2bn` and `test_hex2bn` do a pretty good job stressing the various calls already, so I didn't include anything in `test_test.c`.  Still, I can add some.

I just wanted to make sure there are some tests that guarantee that we get to see what a test failure looks like.  I don't trust the other tests to do so, even if they are good for the moment.

**Comment by levitte** [comment]
@richsalz, changes have been made, please reconfirm?

**Comment by pauli-oracle** [comment]
Seeing sample output is a good reason.  It won't be visible in other tests normally.

I'm toying with the idea of suppressing the output from `test_test` by default to reduce noise in the test output.  The _failure_ output would be enabled using a command line option or somesuch.  I'm not convinced it is a good idea though...

**Comment by richsalz** [comment]
reconfirm.  please merge.

**Comment by levitte** [comment]
> I'm toying with the idea of suppressing the output from `test_test` by default to reduce noise in the test output.  The _failure_ output would be enabled using a command line option or somesuch.  I'm not convinced it is a good idea though...

It's not.  We are, after all, asking for verbosity.  The test harness redirects the recipes' stdout and stderr to /dev/null when running non-verbose, so...

**Comment by levitte** [comment]
Merged.  dc352c193755525292310c8992e3c9b81a556a31 and 03d8e9cb43da5c524e5890a5a51e2c77f1fbd789

**Pull request closed without merging**